keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/schema/Factory.kt

```diff
@@ -0,0 +1,6 @@
+package com.netflix.spinnaker.keel.api.schema
+
+import kotlin.annotation.AnnotationTarget.CONSTRUCTOR
+
+@Target(CONSTRUCTOR)
+annotation class Factory
```

Comment:
Effectively the same thing as `@JsonCreator` without needing to bring in `jackson-annotations`

Reply:
I was debating with myself whether this should be named something more specific, given its purpose... (like `JsonFactory` or `JsonConstructor` perhaps). Probably not worth the energy, but thought I'd mention it.

Reply:
I didn't want anyone to mistake it for an actual Jackson annotation

Reply:
Last try: `SchemaConstructor`? 🙂